### DISCUSSION TITLE
Discriminated Unions on strings only? or maybe it is something else

### DISCUSSION THREAD

**Discussion by jmarbutt** [colinhacks/zod#2943] [Q&A] [answered]
Is it possible to define a `discriminatedUnion` on a type like this:

```js
const type myType = 'type1' | 'type2' | 'type3'
```

How would I use Zod to validate a type is one of the strings above?

**Comment by RenaudAubert** [comment, marked as answer]
I think what you are looking for is [Zod enum](https://zod.dev/?id=zod-enums)
